### PR TITLE
Isolation groups API CRUD

### PR DESCRIPTION
**Pull request by davidporter-id-au**
Adds an interface for updating / reading isolation groups, both at the domain and globally. 

Eg (pseudocode)

```
  client.UpdateGlobalIsolationGroups(ctx, adminv1.UpdateDomainIsolationGroupsRequest{
      domain: "sample-domain",
     // a list of all the zones this is to not be operating in:
      isolation_groups: {
        { 
           name: "zone1",
           state:  adminv1.IsolationGroupStateDrained
        },
      }
  })
```

I wanted to be able to hold-off on an API which is clearly drain oriented or allow-oriented. The IsolationGroup implementation can determine to be deny or allow oriented in its use of this information. 

### Upsert semantics

By a simple GET and SET api I expect callers to add the total state they wish to see every time, this is a complete upsert. 

I expect a subsequent call such as this: 
```
  client.UpdateGlobalIsolationGroups(ctx, adminv1.UpdateDomainIsolationGroupsRequest{
      domain: "sample-domain",
      isolation_groups: {}
  })
```
to remove all drains for this domain

### CODE REVIEW

File: thrift/admin.thrift [hunk 1]
```diff
@@ -265,6 +265,26 @@ service AdminService {
       2: shared.EntityNotExistsError    entityNotExistError,
       3: shared.InternalServiceError    internalServiceError,
     )
+
+  GetGlobalIsolationGroupsRequest GetGlobalIsolationGroups(1: GetGlobalIsolationGroupsResponse request)
```
Comment (Shaddoll) [inline thrift/admin.thrift:269]: response <-> request

File: proto/uber/cadence/admin/v1/service.proto [hunk 2]
```diff
@@ -435,3 +447,33 @@ message DynamicConfigFilter {
 	string name = 1;
 	api.v1.DataBlob value = 2;
 }
+
+// Global
+message GetGlobalIsolationGroupsRequest{}
+
+message GetGlobalIsolationGroupsResponse{
+    repeated IsolationGroupConfiguration isolation_groups = 1;
```
Comment (Shaddoll) [inline proto/uber/cadence/admin/v1/service.proto:455]: should `repeated` be removed?

File: proto/uber/cadence/admin/v1/service.proto [hunk 3]
```diff
@@ -121,6 +121,18 @@ service AdminAPI {
 
   // MaintainCorruptWorkflow deletes a workflow if its history is corrupt due to underlying DB issues (e.g. Cassandra resurrections)
   rpc MaintainCorruptWorkflow(MaintainCorruptWorkflowRequest) returns (MaintainCorruptWorkflowResponse);
+
+  // GetIsolationGroups
+  rpc GetGlobalIsolationGroups(GetGlobalIsolationGroupsRequest) returns (GetGlobalIsolationGroupsResponse);
```
Comment (Shaddoll) [inline proto/uber/cadence/admin/v1/service.proto:126]: Who will call the `Get` APIs?

File: proto/uber/cadence/admin/v1/service.proto (hunk 2 above)
Comment (davidporter-id-au) [reply proto/uber/cadence/admin/v1/service.proto:455]: yes, thank you, I didn't notice I'd done that twice

File: thrift/admin.thrift (hunk 1 above)
Comment (davidporter-id-au) [reply thrift/admin.thrift:269]: thank you

File: proto/uber/cadence/admin/v1/service.proto (hunk 3 above)
Comment (davidporter-id-au) [reply proto/uber/cadence/admin/v1/service.proto:126]: CLI / SRE tooling. I doubt it's possible to have such an endpoint to add drains without being able to read also

**Review by Shaddoll — APPROVED** [review]

**Pull request merged**